### ISSUE TITLE
Should ignoring SSL in local emulator via docker with AZURE_COSMOS_EMULATOR_IP_ADDRESS_OVERRIDE still work?

### ISSUE THREAD

**Issue by travis-icorein** [Azure/azure-cosmos-dotnet-v3#2623]
>We are continuously addressing and improving the SDK, if possible, make sure the problem persist in the [latest SDK version](https://www.nuget.org/packages/Microsoft.Azure.Cosmos).

**Describe the bug**
I see [this workaround ](https://github.com/Azure/azure-cosmos-dotnet-v3/issues/1232#issuecomment-670673733) is available for local dev work and CI/CD.  However, should this still be expected to work with a docker-compose yaml which specifies an IP for `AZURE_COSMOS_EMULATOR_IP_ADDRESS_OVERRIDE`?  I am still getting this error...

> Unhandled exception. Microsoft.Azure.Cosmos.CosmosException : Response status code does not indicate success: InternalServerError (500); Substatus: 0; ActivityId: 00000000-0000-0000-0000-000000000000; Reason: (The remote certificate is invalid because of errors in the certificate chain: UntrustedRoot);
>  ---> System.Security.Authentication.AuthenticationException: The remote certificate is invalid because of errors in the certificate chain: UntrustedRoot

...with dotnet v3 libraries versions: [3.12.0](https://github.com/Azure/azure-cosmos-dotnet-v3/releases/tag/3.12.0) and [3.20.1](https://github.com/Azure/azure-cosmos-dotnet-v3/releases/tag/3.20.1) (latest)

**To Reproduce**
docker-compose.yaml
```
services:

  cosmosdb:
    container_name: cosmosdb
    image: "mcr.microsoft.com/cosmosdb/linux/azure-cosmos-emulator"
    tty: true
    restart: unless-stopped
    mem_limit: 3G
    cpu_count: 2
    environment:
      - AZURE_COSMOS_EMULATOR_PARTITION_COUNT=10
      - AZURE_COSMOS_EMULATOR_ENABLE_DATA_PERSISTENCE=true
      - AZURE_COSMOS_EMULATOR_IP_ADDRESS_OVERRIDE=10.0.0.5 # whole reason network config, so this can be hardcoded
    ports:
      - "8081:8081"  # data explorer ui
      - "10251:10251"
      - "10252:10252"
      - "10253:10253"
      - "10254:10254"
    networks:
      my_net:
        ipv4_address: "10.0.0.5"
    volumes:
      - ./db/cosmos:/data/db:rw


networks:
  my_net:
    driver: bridge
    ipam:
      config:
        - subnet: "10.0.0.0/8"
          gateway: "10.0.0.1"
```

Clone [this repo](https://github.com/lokijota/CosmosDbDeleteDocumentsByQuery) (to backup/delete cosmos data).  Exception mentioned will occur [here](https://github.com/lokijota/CosmosDbDeleteDocumentsByQuery/blob/main/CosmosDBDeleteByQuery/CosmosDBDeleteByQuery/Program.cs#L50).

**Expected behavior**
Successful connection to local Cosmos Llinux Emulator

**Actual behavior**
Exception noted

**Environment summary**
SDK Version: 3.12 & 3.20
OS Version (e.g. Windows, Linux, MacOSX): Windows 10

**Additional context**

> Unhandled exception. Microsoft.Azure.Cosmos.CosmosException : Response status code does not indicate success: InternalServerError (500); Substatus: 0; ActivityId: 00000000-0000-0000-0000-000000000000; Reason: (The remote certificate is invalid because of errors in the certificate chain: UntrustedRoot);
>  ---> System.Security.Authentication.AuthenticationException: The remote certificate is invalid because of errors in the certificate chain: UntrustedRoot
>    at System.Net.Security.SslStream.SendAuthResetSignal(ProtocolToken message, ExceptionDispatchInfo exception)
>    at System.Net.Security.SslStream.ForceAuthenticationAsync[TIOAdapter](TIOAdapter adapter, Boolean receiveFirst, Byte[] reAuthenticationData, Boolean isApm)
>    at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore(Boolean async, Stream stream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)
>    --- End of inner exception stack trace ---
>    at System.Net.Security.SslStream.SendAuthResetSignal(ProtocolToken message, ExceptionDispatchInfo exception)
>    at System.Net.Security.SslStream.ForceAuthenticationAsync[TIOAdapter](TIOAdapter adapter, Boolean receiveFirst, Byte[] reAuthenticationData, Boolean isApm)
>    at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore(Boolean async, Stream stream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)
> --- Cosmos Diagnostics ---{"DiagnosticVersion":"2","Summary":{"StartUtc":"2021-07-17T04:25:10.1822065Z","RunningElapsedTimeInMs":20.9131,"UserAgent":"cosmos-netstandard-sdk/3.12.0|3.11.4|03|X64|Microsoft Windows 10.0.19043|.NET 5.0.7|","TotalRequestCount":0,"FailedRequestCount":0},"Context":[]}

**Comment by travis-icorein** [comment]
I advise holding off on additional investigation/feedback at the moment (details to follow, can't delete "issue")

**Comment by j82w** [comment]
It's recommended is to use the [documented workaround](https://github.com/Azure/azure-cosmos-dotnet-v3/issues/1232#issuecomment-670673733).

**Comment by travis-icorein** [comment]
> I advise holding off on additional investigation/feedback at the moment (details to follow, can't delete "issue")

I found a bug in my implementation of said workaround (already in use), hence I advised holding off on feedback.  I had hoped it would then "just work" (and expected to report back as much).  

However, after running for around an hour (in the scenario/configuration noted in original issue/post above)... the following exception and diagnostic log was returned...

> System.IO.IOException: Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request.

See attached file for full diag log.
[exception-diag.txt](https://github.com/Azure/azure-cosmos-dotnet-v3/files/6859550/exception-diag.txt)
